src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -176,11 +176,9 @@ public static FormulaValue Float(FormattingInfo formatInfo, IRContext irContext,
                 {
                     return CommonErrors.BadLanguageCode(irContext, cultureArg.Value);
                 }
-
-                formatInfo.CultureInfo = culture;
             }
 
-            bool isValue = TryFloat(formatInfo, irContext, args[0], out NumberValue result);
+            bool isValue = TryFloat(new FormattingInfo(culture, formatInfo.TimeZoneInfo), irContext, args[0], out NumberValue result);

```

Comment:
>new FormattingInfo(culture, formatInfo.TimeZoneInfo) [](http://example.com/codeflow?start=36&length=52)

This is cloning the incoming formatInfo, bug adjusting the Culture? 

A pattern here is to wrap that in a helper method like:
formatInfo.With(culture); 


that ensures that we're copying the other fields - which will be more interesting if we add more fields.

Reply:
updated.